hack/tools/fetch-images/main.go

```diff
@@ -0,0 +1,268 @@
+			if err != nil {
+				return nil, err
+			}
+			awsImages, found, err := unstructured.NestedStringMap(values, "hooks", "ccm", "aws", "k8sMinorVersionToCCMVersion")
+			if !found {
+				return images, fmt.Errorf("failed to find k8sMinorVersionToCCMVersion from file %s",
+					path.Join(carenChartDirectory, "values.yaml"))
+			}
+			if err != nil {
+				return images, fmt.Errorf("failed to get map k8sMinorVersionToCCMVersion with error %w",
+					err)
+			}
+			for _, tag := range awsImages {
+				info.stringValues = []string{
+					fmt.Sprintf("image.tag=%s", tag),
+				}
+				chartImages, err := getImagesForChart(info)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get images for %s with error %w", info.name, err)
+				}
+				images = append(images, chartImages...)
+			}
+			// skip the to next addon because we got what we needed
+			continue
+		}
+		chartImages, err := getImagesForChart(info)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get images for %s with error %w", info.name, err)
+		}
+		images = append(images, chartImages...)
+	}
+	return images, nil
+}
+
+func getHelmValues(carenChartDirectory string) (map[string]interface{}, error) {
+	values := path.Join(carenChartDirectory, "values.yaml")
+	valuesFile, err := os.Open(values)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file %s with %w", values, err)
+	}
+	defer valuesFile.Close()
+	m := make(map[string]interface{})
+	err = yaml.NewYAMLOrJSONDecoder(valuesFile, 1024).Decode(&m)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file %s with %w", values, err)
+	}
+	return m, nil
+}
+
+func getValuesFileForChartIfNeeded(chartName, carenChartDirectory string) string {
+	switch chartName {
+	case "nutanix-csi-storage":
+		return path.Join(carenChartDirectory, "addons", "csi", "nutanix", defaultHelmAddonFilename)
+	case "node-feature-discovery":
+		return path.Join(carenChartDirectory, "addons", "nfd", defaultHelmAddonFilename)
+	case "snapshot-controller":
+		return path.Join(carenChartDirectory, "addons", "csi", "snapshot-controller", defaultHelmAddonFilename)
+	case "cilium":
+		return path.Join(carenChartDirectory, "addons", "cni", "cilium", defaultHelmAddonFilename)
+	// this uses the values from kustomize because the file at addons/cluster-autoscaler/values-template.yaml
+	// is a file that is templated
+	case "cluster-autoscaler":
+		return path.Clean(path.Join(carenChartDirectory, "..", "..",
+			"hack", "addons", "kustomize", "cluster-autoscaler", "manifests", "helm-values.yaml"))
+	default:
+		return ""
+	}
+}
+
+func getImagesForChart(info *ChartInfo) ([]string, error) {
+	images := pkg.Images{}
+	images.SetChart(info.name)
+	if info.repo != "" {
+		images.RepoURL = info.repo
+	}
+	if info.valuesFile != "" {
+		images.ValueFiles.Set(info.valuesFile)
+	}
+	if len(info.stringValues) > 0 {
+		images.StringValues = info.stringValues
+	}
+	// kubeVersion needs to be set for some addons
+	images.KubeVersion = "v1.29.0"
```

Comment:
Does this need to change over time? If so, should the command caller pass it as a flag?

Reply:
there was a minimum version required for an app.